ncm-ntpd/src/main/pan/components/ntpd/schema.pan

```diff
@@ -96,6 +96,14 @@ type ntpd_system_options = {
     "stats" ? boolean
 };
 
+@documentation{
+    Defines a single rule for the "interface" statement. See the ntp_misc manpage.
+}
+type ntpd_interface_options = {
+    "action" : choice("listen", "ignore", "drop")
+    "match" : string
```

Comment:
So I read the [ntp_misc manpage](https://www.unix.com/man-page/centos/5/ntp_misc/), the match part seems to have some pretty specific restrictions which are not at all covered by `string`:
>     interface [listen | ignore | drop] [all | ipv4 | ipv6 | wildcard | name | address[/prefixlen]]

What is not clear is which are values and which are placeholders, and then `name` appears to mean network interface name and that is a whole can of worms, but maybe something like the following would cover it?
```suggestion
    "match" : string_trimmed with {
        match(SELF, 'all|ipv4|ipv6|wildcard') || is_ip || is_ipv4_netmask_pair(SELF) || is_ipv6_network_block(SELF) || path_exists(format('/system/network/interfaces/%s', SELF));
    }
```

Reply:
Who needs documentation when you have source code?
https://github.com/ntp-project/ntp/blob/9c75327c3796ff59ac648478cd4da8b205bceb77/ntpd/complete.conf.in#L57-L63

Reply:
Opened an issue (#1481) to look at this in future.